### ISSUE TITLE
Delete event on parent or last child in group

### ISSUE THREAD

**Issue by tspoke** [FrangSierra/Rx2Firebase#39]
Hi @FrangSierra ,

I'm using your library and I got an issue when a parent is deleted.

Let's say I have this structure in my firebase database : 

```
posts: {
    my_group: {
         key_1: true,
         key_2: true
    }
}
```

If I observeValueEvent on `my_group`, if a node is deleted it works. But, if the last node is deleted the onNext is never called (because the key `my_group` has been deleted by firebase). The same thing happen if the `posts` or directly the `my_group` key are deleted (even if there is child in it).

An event is triggered, but there is no data in it so the `onNext` is never called. From your code : 

```
final ValueEventListener valueEventListener = new ValueEventListener() {
    @Override
    public void onDataChange(final DataSnapshot dataSnapshot) {
            if (dataSnapshot.exists()) {
                emitter.onNext(dataSnapshot);
            } else {
                // Here, we need to dispatch the info because the root node (or all child) have been potentially deleted.
            }
     }
     ...
};
```

Do you have an idea to handle this case ?

**Edit:**

If I remove the `dataSnapshot.exists()` condition in your code, I get the expected behavior (empty list).

**Comment by FrangSierra** [comment]
I just released today the version 1.4.0. The fix for this issue is already there. Enjoy it!

**Comment by tspoke** [comment]
Long life to you @FrangSierra ! \o/
